File: 76 Russian Roulette/python/russianroulette.py
from random import random

NUMBER_OF_ROUNDS = 9


def initial_message():
    print(" " * 28 + "Russian Roulette")
    print(" " * 15 + "Creative Computing  Morristown, New Jersey\n\n\n")
    print("This is a game of >>>>>>>>>>Russian Roulette.\n")
    print("Here is a Revolver.")


def parse_input():
    correct_input = False
    while not correct_input:
        try:
            i = int(input('?'))
            correct_input = True
        except ValueError:
            print('Number expected...')
    return i


initial_message()
while True:
    dead = False
    n = 0
    print("Type \'1\' to Spin chamber and pull trigger")
    print("Type \'2\' to Give up")
    print("Go")
    while not dead:
        i = parse_input()

        if i == 2:
            break

        if random() > 0.8333333333333334:
            dead = True
        else:
            print("- CLICK -\n")
            n += 1

        if n > NUMBER_OF_ROUNDS:
            break
    if dead:
        print("BANG!!!!!   You're Dead!")
        print("Condolences will be sent to your relatives.\n\n\n")
        print("...Next victim...")
    else:
        if n > NUMBER_OF_ROUNDS:
            print("You win!!!!!")
            print("Let someone else blow his brain out.")
        else:
            print("     Chicken!!!!!\n\n\n")
            print("...Next victim....")

File: 76 Russian Roulette/perl/russianroulette.pl
#!/usr/bin/perl
#use strict;
# Automatic converted by bas2perl.pl

print ' 'x28 . "RUSSIAN ROULETTE\n";
print ' 'x15 . "CREATIVE COMPUTING MORRISTOWN, NEW JERSEY\n";
print "\n"; print "\n"; print "\n";
print "THIS IS A GAME OF >>>>>>>>>>RUSSIAN ROULETTE.\n";
Line10:
print "\n"; print "HERE IS A REVOLVER.\n";
Line20:
print "TYPE '1' TO SPIN CHAMBER AND PULL TRIGGER.\n";
print "TYPE '2' TO GIVE UP.\n";
print "GO";
$N=0;
Line30:
print "? "; chomp($I = <STDIN>);
if ($I ne 2) { goto Line35; }
print " CHICKEN!!!!!\n";
goto Line72;
Line35:
$N=$N+1;
if (rand(1)>.833333) { goto Line70; }
if ($N>10) { goto Line80; }
print "- CLICK -\n";
print "\n"; goto Line30;
Line70:
print " BANG!!!!! YOU'RE DEAD!\n";
print "CONDOLENCES WILL BE SENT TO YOUR RELATIVES.\n";
Line72:
print "\n"; print "\n"; print "\n";
print "...NEXT VICTIM...\n"; goto Line20;
Line80:
print "YOU WIN!!!!!\n";
print "LET SOMEONE ELSE BLOW HIS BRAINS OUT.\n";
goto Line10;
exit;


